src/components/thumbnail/document-collection-list.tsx

```diff
@@ -12,18 +13,53 @@ interface IProps {
   subTab: ISubTabSpec;
   tabSpec: NavTabModelType;
   selectedDocument?: string;
+  horizontal?: boolean;
+  collapsed?: boolean;
+  scrollToLocation?: number;
   onSelectNewDocument?: (type: string) => void;
   onSelectDocument?: (document: DocumentModelType) => void;
+  setScrollWidth?: (scrollWidth: number) => void;
+  setScrollLeft?: (scrollLeft: number) => void;
 }
 
 export const kNavItemScale = 0.11;
 
 export const DocumentCollectionList: React.FC<IProps> = observer(function DocumentCollectionList(
-    { subTab, tabSpec, selectedDocument, onSelectNewDocument, onSelectDocument }) {
+    { subTab, tabSpec, horizontal, collapsed, selectedDocument, scrollToLocation,
+      onSelectNewDocument, onSelectDocument, setScrollWidth,  setScrollLeft}) {
   const ui = useUIStore();
   const appConfigStore = useAppConfig();
   const user = useUserStore();
   const navTabSpec = appConfigStore.navTabs.getNavTabSpec(tabSpec.tab);
+  const documentListRef = useRef<HTMLDivElement>(null);
+  useEffect(()=>{
+    const documentListEl = documentListRef.current;
+    if (documentListEl) {
+      setScrollWidth && setScrollWidth(documentListEl.scrollWidth);
+      setScrollLeft && setScrollLeft(documentListEl.scrollLeft);
+    }
+  }, [documentListRef, setScrollLeft, setScrollWidth]);
+
+  const handleBrowserScroll = useCallback((documentListEl: HTMLDivElement) => (evt: any) => {
+    setScrollWidth && setScrollWidth(documentListEl.scrollWidth);
+    setScrollLeft && setScrollLeft(documentListEl.scrollLeft);
+  },[setScrollLeft, setScrollWidth]);
+
+  useEffect(()=>{
+    const documentListEl = documentListRef.current;
+    if (documentListEl) {
+      documentListEl.addEventListener("scroll", handleBrowserScroll(documentListEl));
+    }
+    return () => {
+      documentListEl?.removeEventListener("scroll",  handleBrowserScroll(documentListEl));
+    };
+  },[handleBrowserScroll]);
+
+  useEffect(() => {
+    if(scrollToLocation) {
+      documentListRef.current?.scroll({left: scrollToLocation, behavior: "smooth"});
+    }
+  },[scrollToLocation]);
```

Comment:
I think that if you use `scrollTo` here instead of `scroll`, you can take my advice above about what `scrollToLocation` contains (namely, the pixel location you want to scroll to, rather than the amount you want to scroll by).